force-di/main/classes/di_Injector.cls

```diff
@@ -91,6 +91,45 @@ public class di_Injector {
         return bindings[0].getInstance(params);
     }
 
+    /**
+     * Creates an instance of the object configured for the named binding and the binding SObjectType
+     **/
+    public Object getInstance(Type developerNameByType, Schema.SObjectType bindingSObjectType) {
+        return getInstance(developerNameByType, bindingSObjectType, null);
+    }
+
+    /**
+     * Creates an instance of the object configured for the named binding and the binding SObjectType
+     *   (if params is specified, implicitly uses the Provider interface)
+     **/
+    public Object getInstance(Type developerNameByType, Schema.SObjectType bindingSObjectType, Object params) {
+        return getInstance(developerNameByType == null ? null : developerNameByType.getName().toLowerCase(), bindingSObjectType, params);
```

Comment:
@ImJohnMDaniel I think this method can pass `developerNameByType` as-is without the null check or changing case as that logic is handled at the point of use at lines 114 and 123. Removing this logic would reduce the places it's spread throughout the code.

Reply:
@douglascayers -- I am going to leave the null check in place on this line.  It is needed because that line accesses the `System.Type.getName()` method on the Type pararm developerNameByType.  An NPE could surface on line 106 if the null check were not performed.

Reply:
@ImJohnMDaniel good point, sorry, I thought it was just lowercasing a string, I missed the `.getName()` method call